### DISCUSSION TITLE
Pass an option to wire.callTableColumnAction() from a custom table column component

### DISCUSSION THREAD

**Discussion by sinnbeck** [filamentphp/filament#7666] [Help] [answered]
I am working on a small table column component for downloading files that are on a relationship. For this I use the dropdown blade component. I would I to use an action to handle the redirect, when an option is selected. Sadly I can only pass the name and the recordKey to the `callTableColumnAction()` method. Is there a way that I can set the $state or similar to use an action?

For now I am just using the `updateTableColumnState()` as that allows me to pass in the ID of the file to download, and then use `updateStateUsing()` in the form, to do the download (redirect).

This is my current blade component code
```blade
<div
        x-data="{
        error: undefined,

        isLoading: false,

        name: @js($getName()),

        recordKey: @js($recordKey),

        state: @js($state),
    }"
>
    @if(count($getOptions()))
        <x-filament::dropdown placement="bottom-start">
            <x-slot name="trigger">
                <x-filament::button :color="$getColor()">
                    {{$getLabel()}}
                </x-filament::button>
            </x-slot>
            @foreach ($getOptions() as $key => $option)
                <x-filament::dropdown.list.item x-on:click="
            isLoading = true

            const response = await $wire.updateTableColumnState(
                name,
                recordKey,
                '{{$key}}',
            )

            error = response?.error ?? undefined

            isLoading = false
        ">
                    {{$option}}
                </x-filament::dropdown.list.item>
            @endforeach

        </x-filament::dropdown>
    @else
        No files
    @endif
</div>

**Comment by sinnbeck** [comment]
Another idea I had was to use an `ActionGroup` to in which I would loop over the files, but sadly it only accepts an array, and not a closure that returns an array

**Comment by danharrin** [comment]
I would not use callTableColumnAction or updateTableColumnState, define a public method on the List page or Livewire component and use Livewire as normal. Those methods weren't designed for this

**Reply by sinnbeck** [reply]
Ok will do. But if I later want to make a "dropdown" plugin, is there a way to pass the item to the action ?

Alternatively I will see if I can make a wrapper for ActionGroup instead, that does the same :)

**Reply by danharrin** [reply, marked as answer]
Do you really need to be able to interact with Livewire like this? Or can you preload the link into the HTML 🤔 

One option is getting plugin users to add a trait to their Livewire components with your methods on it

**Reply by sinnbeck** [reply]
The idea with adding a trait + an interface is actually very good :) Thanks again
